### PR TITLE
Feat/custom head meta

### PR DESCRIPTION
**Pull request by jamiemonksuk**

### CODE REVIEW

File: src/sprout/Helpers/CustomHeadTags.php [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        foreach ($tags as &$tag) {
+            $tag['attr_values'] = json_decode($tag['attr_values'], true);
+        }
+        unset($tag);
+
+        return $tags;
+    }
+
+
+    public static function renderTagsFormElementHome(int $homepage_id)
```
Comment (gwillz) [inline src/sprout/Helpers/CustomHeadTags.php:200]: Missing docs here.

File: src/sprout/Helpers/CustomHeadTags.php [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+{
+
+    /**
+     * A list of possible tag combinations in the format of tag_type => [tags] => [values]
+     *
+     * NOTE: This can be customised by overriding the same setup in Kohana::config('sprout.custom_meta.available_tag_list')
+     *
+     * @var array
+     */
+    protected static $available_tag_list = [
```
Comment (gwillz) [inline src/sprout/Helpers/CustomHeadTags.php:31]: I recall kohana configs are supposed to have this sort of 'default/override' behaviour built-in. But I suspect it works differently for the sprout config. Shame.

File: src/sprout/Helpers/CustomHeadTags.php [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+        ],
+    ];
+
+
+    /**
+     * Get the list of available tags, allowing override form config if set
+     *
+     * @return array
+     */
+    public static function getAvailableTagList()
```
Comment (gwillz) [inline src/sprout/Helpers/CustomHeadTags.php:109]: We can add strict types here, which should save our butts in future when we inevitably install an invalid config.

```public static function getAvailableTagList(): array```

Same for all the methods in this helper. With a minimum php 7.4 we can also use nullish and void types.

Comment (jamiemonksuk) [reply src/sprout/Helpers/CustomHeadTags.php:109]: Good thinking, doing this now

File: src/sprout/Helpers/CustomHeadTags.php (hunk 2 above)
Comment (jamiemonksuk) [reply src/sprout/Helpers/CustomHeadTags.php:31]: Nah maybe you're right, we can add a core config for it, that a user config would override

**Comment by gwillz** [comment]
@jamiemonksuk you've merged this elsewhere, yes?

**Comment by jamie-kb** [comment]
That is correct :)

It came in with the social meta rework, this can be closed!

--
senior developer
www.karmabunny.com.au


talk // 08 7120 7100
post // po box 188, stepney sa 5069

**Pull request closed without merging**